app/models/binary/DataSet.scala

```diff
@@ -321,20 +320,19 @@ class DataSetResolutionsDAO @Inject()(sqlClient: SQLClient)(implicit ec: Executi
   def updateResolutions(_dataSet: ObjectId, dataLayersOpt: Option[List[DataLayer]]): Fox[Unit] = {
     val clearQuery = sqlu"delete from webknossos.dataSet_resolutions where _dataSet = ${_dataSet.id}"
     val insertQueries = dataLayersOpt match {
-      case Some(dataLayers: List[DataLayer]) => {
-        dataLayers.map { layer =>
+      case Some(dataLayers: List[DataLayer]) =>
+        dataLayers.flatMap { layer =>
           layer.resolutions.map { resolution =>
             {
               sqlu"""insert into webknossos.dataSet_resolutions(_dataSet, dataLayerName, resolution)
                        values(${_dataSet.id}, ${layer.name}, '#${writeStructTuple(resolution.toList.map(_.toString))}')"""
             }
           }
-        }.flatten
-      }
+        }
       case _ => List()
     }
     for {
-      _ <- run(DBIO.sequence(List(clearQuery) ++ insertQueries).transactionally)
+      _ <- run(DBIO.sequence(List(clearQuery) ++ insertQueries).transactionally.withTransactionIsolation(Serializable))
```

Comment:
I believe anything that is run as `Serializable` aborts on conflict, so we should probably add a retry around it, as in the other places where this happens.